core/src/main/java/org/infinispan/metrics/impl/MetricsComponentFactory.java

```diff
@@ -74,20 +78,47 @@ private static void logMissingDependencies(Throwable t) {
       log.debug("Micrometer metrics are not available because classpath dependencies are missing.", t);
    }
 
-   private synchronized MetricsRegistry createMetricRegistry() {
+   private synchronized MetricsRegistry createMetricRegistry(ClassLoader classLoader) {
       if (registry != null) {
          return registry;
       }
       if (isMetricsDisabled()) {
          registry = NoMetricRegistry.NO_OP_INSTANCE;
          return registry;
       }
+
+      String registryClass = "io.micrometer.prometheusmetrics.PrometheusMeterRegistry";
       try {
-         registry = new MetricsRegistryImpl();
-      } catch (Throwable t) {
-         logMissingDependencies(t);
-         registry = NoMetricRegistry.NO_OP_INSTANCE;
+         loadClassStrict(registryClass, classLoader);
+         registry = new PrometheusRegistry();
+      } catch (ClassNotFoundException ignore) {
+         logMissingMicrometerImpl(registryClass);
+         try {
+            registryClass = "io.micrometer.prometheus.PrometheusMeterRegistry";
+            loadClassStrict(registryClass, classLoader);
+            registry = new PrometheusSimpleClientRegistry();
+         } catch (ClassNotFoundException e) {
+            logMissingMicrometerImpl(registryClass);
+
+            MicrometerMeterRegistryConfiguration configuration = globalConfig.module(MicrometerMeterRegistryConfiguration.class);
+            MeterRegistry configuredRegistry = configuration == null ? null : configuration.meterRegistry();
+            if (configuredRegistry != null) {
+               log.debugf("Fallback to %s for provided MeterRegistry '%s'", NoMetricRegistry.class.getSimpleName(), configuredRegistry.getClass().getName());
+               registry = new AbstractMetricsRegistry() {
+                  @Override
+                  ScrapeRegistry createScrapeRegistry(MeterRegistry registry) {
+                     return new NoScrapeRegistry(registry, true);
+                  }
+               };
+            } else {
+               registry = NoMetricRegistry.NO_OP_INSTANCE;
+            }
```

Comment:
I'm having second thoughts about this. It cannot be here as `MetricsComponentFactory` cannot import anything from micrometer. Let's remove it.